bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -861,7 +875,21 @@ private void writeAndFlush(final Channel channel,
         }
 
         try {
-            channel.writeAndFlush(request, channel.voidPromise());
+            final long startTime = MathUtils.nowInNano();
+            ChannelFuture future = channel.writeAndFlush(request, channel.voidPromise());
```

Comment:
Is this working? With voidPromise?

@merlimat @ivankelly I think not using voidPromise we are going to add an allocation on the hot path

Reply:
yes, the void promise bit is there to avoid an allocation. it was part of yahoo's GC changes.

Reply:
with `voidPromise` the returned future will be the one from the `voidPromise`, so not very useful :)